### ISSUE TITLE
ValueError: Convolution expects input with rank 4, got 1

### ISSUE THREAD

**Issue by andreqwert** [sanku-lib/image_triplet_loss#3]
Hello!
Thanks for sharing your project.
I tried to run training with my own dataset but the following error was occured:
```
Dataset loaded successfully.
Preprocessing Done. Summary:
Images train : (837,)
Labels train : (837,)
Images test  : (91,)
Labels test  : (91,)
Unique label : [ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
 24 25 26 27 28 29]
placeholder_shape [None]
WARNING:tensorflow:From train_triplets.py:25: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.

WARNING:tensorflow:From /home/user/similarity/sanku/model.py:8: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.

WARNING:tensorflow:
The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
  * https://github.com/tensorflow/io (for I/O related ops)
If you depend on functionality not listed there, please file an issue.

Traceback (most recent call last):
  File "train_triplets.py", line 30, in <module>
    anchor_output = model.conv_net(anchor_input, reuse=False)
  File "/home/user/similarity/sanku/model.py", line 11, in conv_net
    scope=scope, reuse=reuse)
  File "/home/user/.local/lib/python3.6/site-packages/tensorflow/contrib/framework/python/ops/arg_scope.py", line 182, in func_with_args
    return func(*args, **current_args)
  File "/home/user/.local/lib/python3.6/site-packages/tensorflow/contrib/layers/python/layers/layers.py", line 1159, in convolution2d
    conv_dims=2)
  File "/home/user/.local/lib/python3.6/site-packages/tensorflow/contrib/framework/python/ops/arg_scope.py", line 182, in func_with_args
    return func(*args, **current_args)
  File "/home/user/.local/lib/python3.6/site-packages/tensorflow/contrib/layers/python/layers/layers.py", line 1025, in convolution
    (conv_dims + 2, input_rank))
ValueError: Convolution expects input with rank 4, got 1
```
Please help. What should I improve for a launch? 

My packages versions are all from requirements.txt. Besides, I tried to run the program without any env (tensorflow 1.14, numpy 1.18) but got the same error.

**Comment by sanku-lib** [comment]
Hi @andreqwert ,

The image shape should be [ X, n, m, 3 ] 
where X: number of images
Dimension of image: n*m
as it's rgb image use 3.

So, just reshape your training and test image to proper format it should work. 
Once you are done with that, please verify if placeholder_shape is as follows:
placeholder_shape [None, 28, 28, 3]

My log looks like something as follows:

Dataset loaded successfully.
Preprocessing Done. Summary:
Images train : (26999, 28, 28, 3)
Labels train : (26999,)
Images test  : (2998, 28, 28, 3)
Labels test  : (2998,)
Unique label : [0 1 2 3 4 5]
placeholder_shape [None, 28, 28, 3]


Hope this helps, please let me know if there is any more issue.

Thanks,
Shibsankar

**Comment by piyushsingh0395** [comment]
> Hello!
> Thanks for sharing your project.
> I tried to run training with my own dataset but the following error was occured:
> 
> ```
> Dataset loaded successfully.
> Preprocessing Done. Summary:
> Images train : (837,)
> Labels train : (837,)
> Images test  : (91,)
> Labels test  : (91,)
> Unique label : [ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
>  24 25 26 27 28 29]
> placeholder_shape [None]
> WARNING:tensorflow:From train_triplets.py:25: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.
> 
> WARNING:tensorflow:From /home/user/similarity/sanku/model.py:8: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.
> 
> WARNING:tensorflow:
> The TensorFlow contrib module will not be included in TensorFlow 2.0.
> For more information, please see:
>   * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
>   * https://github.com/tensorflow/addons
>   * https://github.com/tensorflow/io (for I/O related ops)
> If you depend on functionality not listed there, please file an issue.
> 
> Traceback (most recent call last):
>   File "train_triplets.py", line 30, in <module>
>     anchor_output = model.conv_net(anchor_input, reuse=False)
>   File "/home/user/similarity/sanku/model.py", line 11, in conv_net
>     scope=scope, reuse=reuse)
>   File "/home/user/.local/lib/python3.6/site-packages/tensorflow/contrib/framework/python/ops/arg_scope.py", line 182, in func_with_args
>     return func(*args, **current_args)
>   File "/home/user/.local/lib/python3.6/site-packages/tensorflow/contrib/layers/python/layers/layers.py", line 1159, in convolution2d
>     conv_dims=2)
>   File "/home/user/.local/lib/python3.6/site-packages/tensorflow/contrib/framework/python/ops/arg_scope.py", line 182, in func_with_args
>     return func(*args, **current_args)
>   File "/home/user/.local/lib/python3.6/site-packages/tensorflow/contrib/layers/python/layers/layers.py", line 1025, in convolution
>     (conv_dims + 2, input_rank))
> ValueError: Convolution expects input with rank 4, got 1
> ```
> 
> Please help. What should I improve for a launch?
> 
> My packages versions are all from requirements.txt. Besides, I tried to run the program without any env (tensorflow 1.14, numpy 1.18) but got the same error.

resize the image 28*28
